src/main/kotlin/com/wafflestudio/ggzz/domain/letter/service/LetterService.kt

```diff
@@ -49,15 +51,19 @@ class LetterService(
     fun getLetters(pos: Pair<Double, Double>, range: Int): ListResponse<Response> {
         return ListResponse(letterRepository.findAll().filter { letter ->
             val letterPos = letter.longitude to letter.latitude
-            distanceBetweenTwoPositionInMeter(pos, letterPos) <= range
+            distanceBetweenTwoPositionInMeter(pos, letterPos) <= range && letter.isViewable()
         }.map { Response(it) })
     }
 
     fun getLetter(id: Long, pos: Pair<Double, Double>): DetailResponse {
         val letter = letterRepository.findLetterById(id) ?: throw LetterNotFoundException(id)
+
+        if (!letter.isViewable()) {
+            throw LetterViewableTimeExpiredException()
+        }
         val letterPos = letter.longitude to letter.latitude
-        if (distanceBetweenTwoPositionInMeter(pos, letterPos) > CONCEDE)
-            throw LetterNotCloseEnoughException()
+        if (letter.viewRange != 0 && distanceBetweenTwoPositionInMeter(pos, letterPos) > letter.viewRange)
+            throw LetterNotCloseEnoughException(letter.viewRange)
```

Comment:
해당 Exception의 String explanation을 보면, "... within ${letter.viewRange}"인데 맥락상 아래처럼 인자를 주는 것이 자연스럽지 않을까 합니다!

```suggestion
            throw LetterNotCloseEnoughException(distanceBetweenTwoPositionInMeter(pos, letterPos) - letter.viewRange)
```

Reply:
제 의도는 몇 미터 더 접근해야 함이 아니고 '끄적 주변 {viewRange}미터 에서 확인 가능' 이라고 전달하고 싶었던건데, 에러 메세지가 조금 모호하면 수정하는게 좋을까요?

Reply:
아하 그 의도셨군요!
그런데 보통 `LetterNotCloseEnoughException`를 사용하는 상황은 _User_ 관점에서 "**letter을 보고자 했으나 멀리 떨어져있기 때문에 볼 수 없음**"이므로, `viewRange`만을 사용하는 로직은 _Letter_ 의 관점에서 접근한 의도가 아닌가 합니다.

Reply:
말씀해주신거 반영해서 에러메세지 수정했습니다!